pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/retention/RetentionManager.java

```diff
@@ -136,10 +150,22 @@ private void manageRetentionForTable(TableConfig tableConfig) {
     if (TableNameBuilder.isOfflineTableResource(tableNameWithType)) {
       manageRetentionForOfflineTable(tableNameWithType, retentionStrategy, untrackedSegmentsDeletionBatchSize);
     } else {
-      manageRetentionForRealtimeTable(tableNameWithType, retentionStrategy, untrackedSegmentsDeletionBatchSize);
+      String rawTableName = TableNameBuilder.extractRawTableName(tableNameWithType);
+      TableConfig offlineTableConfig = _pinotHelixResourceManager.getOfflineTableConfig(rawTableName);
+      // hybrid table check should be performed before the realtime table check.
+      if (_isHybridTableRetentionStrategyEnabled && isHybridTable(tableConfig, offlineTableConfig)) {
```

Comment:
(minor) Can reduce one ZK lookup
```suggestion
      TableConfig offlineTableConfig;
      // hybrid table check should be performed before the realtime table check.
      if (_isHybridTableRetentionStrategyEnabled && (offlineTableConfig = _pinotHelixResourceManager.getOfflineTableConfig(rawTableName)) != null) {
```

Reply:
Checkstyle is complaining when code is written in the above style. Looks like it reduces readability.

```
[WARNING] src/main/java/org/apache/pinot/controller/helix/core/retention/RetentionManager.java:[158,34] (coding) InnerAssignment: Inner assignments should be avoided.
```